src/site/_includes/components/cookie/index.ts

```diff
@@ -0,0 +1,74 @@
+import { getCookie, setCookie } from './cookie'
+
+const COOKIE_BANNER_CLASS_PREFIX = 'js-cookie-guidance-banner'
+const COOKIE_PREFERENCE_KEY = 'cookie_preference'
+
+const getClassName = (className: string) => `.${COOKIE_BANNER_CLASS_PREFIX}--${className}`
+
+type CookieMessageToggles = {
+  show: VoidFunction
+  hide: VoidFunction
+}
+
+const getCookieMessageElements = <T extends readonly string[]>(
+  parentElement: Element,
+  types: T,
+): Record<T[number], CookieMessageToggles> => (
+  types.reduce((acc, type) => {
```

Comment:
Would `types.map()` + `Object.fromEntries()` work for this? Just find `reduce()` bit difficult to follow. Would need to load a polyfill for `Object.fromEntries` though (hopefully can just load it from core-js).